### ISSUE TITLE
Some invalid star expressions do not give syntax errors

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ruff#19077]
### Summary

I noticed some invalid star expressions do not give a `SyntaxError`. I tested all I could think of, but I might have missed some cases. I also tested if they fail in `ast.parse` or only in `exec`, since I think that affects where the `SyntaxError` should be raised from.
https://play.ruff.rs/140067c2-f088-4992-847a-26a6652cc88e
```py
# SyntaxError: Starred expression cannot be used here
(*
*[],)

# No error
# Should give SyntaxError: Invalid star expression
# Fails in ast.parse
print(*
*[])

# No error
# Should give SyntaxError: can't use starred expression here
# Passes ast.parse, fails in exec
f"{*[]}"

# No error
# Should give SyntaxError: starred assignment target must be in a list or tuple
# Passes ast.parse, fails in exec
for *a in []:...
```

### Version

playground

**Comment by ntBre** [comment]
I think the last one is okay because it gets an error when I delete the first one. I guess I'm not quite sure that that's expected (one syntax error interfering with another), but at least we catch it otherwise.

The middle two do seem like bugs. Thanks for checking `ast.parse` vs `exec` too. The first case with `print` should be emitted by the parser as a `ParseError`, and the second should be a `SemanticSyntaxError`.

**Comment by dylwil3** [comment]
We could support literally this example

```python
# No error
# Should give SyntaxError: can't use starred expression here
# Passes ast.parse, fails in exec
f"{*[]}"
```

but it's worth pointing out that CPython also gives a SyntaxError whenever the value of the starred expression is computed to be empty. For example:

```python
f"{*[x for x in range(3) if x > 5]}"
```

At that point we're probably in `ty` territory.
